pkg/compactor/executor.go

```diff
@@ -106,6 +120,17 @@ func (e *schedulerExecutor) run(ctx context.Context, c *MultitenantCompactor) er
 	workerID := fmt.Sprintf("compactor-%s", c.ringLifecycler.GetInstanceID())
 	level.Info(e.logger).Log("msg", "compactor running in scheduler mode", "scheduler_endpoint", e.cfg.SchedulerAddress, "worker_id", workerID)
 
+	// Clean up any stale compaction directories from previous runs.
+	compactDir := c.compactorCfg.DataDir + "/compact"
```

Comment:
This has to be a `filepath.Join`. As an example of why the default `./data-compactor/` ends with a `/`.